core/services/bulletprooftxmanager/eth_broadcaster_test.go

```diff
@@ -249,7 +249,9 @@ func TestEthBroadcaster_ProcessUnstartedEthTxs_Success_OnOptimism(t *testing.T)
 	}
```

Comment:
I just realised, I don't think this correctly covers the case. We assume here that the etx got inserted with the correct gas limit, but the gas limit is actually decided at insert time and we don't have logic for that.

Check:

- orm.IdempotentInsertEthTxTaskRun
- core/services/offchainreporting/transmitter.go:39
- core/services/keeper/orm.go:151
- core/services/fluxmonitorv2/orm.go:115

Reply:
It's already starting to look like we need a separate system to handle Optimism transactions.